File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode curr=new ListNode(0,head);
        ListNode currHead=curr;
        if(head==null)
        {
            return head;
        }
        while(currHead.next!=null)
        {
            if(currHead.val!=val && currHead.next.val==val)
            {
                currHead.next=currHead.next.next;
            }else
            {
                currHead=currHead.next;
            }
        }

        return curr.next;
        
    }
}

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        node1=ListNode(0)
        node1.next=head

        node=node1



        while node!=None and node.next!=None:


            if node.val!=val and node.next.val==val:
                node.next=node.next.next
            else:
                node=node.next


            
        return node1.next
        